src/praisonai-code/praisonai_code/cli/commands/code.py

```
@@ -293,16 +293,17 @@ def code_main(
+    # wrapper-only extras, so use it only when the wrapper is installed and fall
+    # back to the resident TUI otherwise.
     from praisonai_code._wrapper_bridge import wrapper_available
 
     if not wrapper_available():
-        typer.echo(
-            "Error: code requires the praisonai wrapper. "
-            "Install the full wrapper: pip install praisonai",
-            err=True,
-        )
-        raise typer.Exit(1)
+        _run_resident_code(prompt, args, compact=compact, session_id=session_id)
```

Ruff (0.16.1)

[error] Undefined name `compact`

(F821)